addons/udes_security/tests/test_password_manager.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+
+import time
+
+from odoo import fields
+from odoo.exceptions import AccessError
+from odoo.tests import common
+
+
+@common.at_install(False)
+@common.post_install(True)
+class TestPasswordManager(common.TransactionCase):
+    """
+    Check that Password Managers can update the password of other users,
+    including admin/debug users if they are also a debug user.
+    """
+
+    def setUp(self):
+        """Create users for testing"""
+        super(TestPasswordManager, self).setUp()
+        self._setup_test_users()
+        # Amount of seconds to wait before updating password
+        # This is done to ensure the `password_write_date` can be used to verify
+        # if a user's password was updated
+        self.password_wait_interval = 2
+
+    def _setup_test_users(self):
+        """
+        Create the following users:
+
+        1. Password Manager
+        2. Debug User
+        3. Standard User (default permissions)
+        """
+        self.group_password_manager = self.env.ref("udes_security.group_password_manager")
+        self.group_debug_user = self.env.ref("udes_security.group_debug_user")
+
+        self.password_manager = self.create_user("Password Manager", self.group_password_manager)
+        self.debug_user = self.create_user("Debug User", self.group_debug_user)
+        self.standard_user = self.create_user("Standard User")
+
+        self.all_users = self.password_manager | self.debug_user | self.standard_user
+
+    def create_user(self, name, groups=None, **kwargs):
+        """
+        Create and return a user record
+
+        :args:
+            - name: String for the name of the user
+            - groups: Recordset of `res.groups` to apply to the user,
+                      if None only the default groups will be used
+        """
+        User = self.env["res.users"]
+
+        login = name.lower().replace(" ", "_")
+        user_vals = {"name": name, "login": login}
+
+        if groups:
+            group_ids = [group.id for group in groups]
```

Comment:
have you tried groups.ids work here?
It may or may not work just want to check

Reply:
Good spot, thanks, have changed this. :)